### PR TITLE
Pat/Initial_org_and_comments

### PR DESCRIPTION
**Pull request by patmccler**
Proof of concept working for relation between classes
In a nut shell, main.rb creates a Test
That test creates a users and tells it to log in

The user know about the login_page through the pages_controller
The user provides the login_page with its username and pw
The user tells the page to hit the submit button 
(the user knows that it has those actions available, and what info it needs to provide
but it doesn't know how the page is actually laid out or what the things are called)

I'd like to think of a better name for pages_controller.
Maybe Directory? not sure... but pages_controller kept making think it was something other than what it was

No selenium involvement yet, that'll be next!!

Definitely happy to go over this before merge / keep it separate for now / whatever you wanna do
This is how I was picturing things when I drew my chart, I hope actually seeing the code can help explain

### CODE REVIEW

**Comment by patmccler** [comment]
Definitely don't approve this until we get a chance to go over it!

**Comment by andrew-k9** [comment]
In leu of my better judgement two things from giving that a glance:

1. `User` should 100% be a class I agree. My idea of having a SalesGroup is better used as a module. In leu of mansplaining here is an example:

``` ruby
module Sales
  # what a sales rep can do
end

module Mentor
  # access to time an attendance, whatnot
end 

module Director
  #access to a looot
end

class User
  # what’s in there now
end

class RemodelingConsultant < User
  includes Sales
end

class RemodelingConsultantMentor < User
  includes Sales, Mentor
end

class Kraig < User
  includes Mentor, Director
end
```

2. I am *very* against user classes having any knowledge of selenium. A user is interacting w/ the DOM but they don’t know they’re doing it. They just need to know their actions on that page. Ex A driver of a car doesn’t need to know about pistons in a combustable engine to hit the gas pedal and make car go.

If any of this sounds condescending feel free to  hit me tomorrow!

**Comment by andrew-k9** [comment]
@patmccler 100% going to have to look at this w/ you. Difficult following on here because all the changes you did and I get of track easily 😞

**Comment by patmccler** [comment]
My thought for 
```I am very against user classes having any knowledge of selenium. A user is interacting w/ the DOM but they don’t know they’re doing it```
is that if we don't have the user hold the driver, won't they have to re-login every time they want to go to a new page?

IE if I have the driver spawned by the homes page, and then want to check tickets, will I need to spawn a new driver, and therefore re-log in, to visit the new page?

OR, perhaps we can have the driver live in page_controller (Which the user owns, and will make a copy of when we spawn the user)
This way, it'd be one driver PER user, without them knowing how it works.
It could be passed to the PageObject so that the different PageObjects can use the same driver


TLDR: we need a way to have one driver used across the different PageObjects, right?
Is that how you see it?

**Comment by patmccler** [comment]
The way to follow the flow of my stuff:
Start in Main

Main creates a test, and tells it to run

the test creates a user, and tells the user to login

The login action on user knows it needs to use the login page, so it creates one from the PageController (Only has a link through that, doesn't know about the actual LoginPage Object)

LoginObject does the work, with input provided from the User's actions
I'll go over it with you after lunch

**Comment by andrew-k9** [comment]
@patmccler 

> is that if we don't have the user hold the driver, won't they have to re-login every time they want to go to a new page?

I already have it set up like that, on initialization, the specific user object gets an instance of PagesController, which has all the driver information. Then the call `RC.login` would be really calling `@controller.login_page`

Also it was more I don't want to clone this branch locally and looking at the diff makes my eyes hurt

**Comment by andrew-k9** [comment]
We need to clean up comments in this, but that's. later issue. Good for now @patmccler !

**Review by andrew-k9 — APPROVED** [review]
Oohhh this is how todo this.... Uhhhh

**Comment by patmccler** [comment]
Hoooooray! One of many!
